### DISCUSSION TITLE
JSON Formatter Specification

### DISCUSSION THREAD

**Discussion by ataylorme** [eslint/eslint#18137] [Q&A] [answered]
I am programmatically parsing the ESLint [JSON formatter](https://eslint.org/docs/latest/use/formatters/#json) output. So far it has been trial and error to determine what fields are present in the output, what is optional, etc.

Is there a specification available for the JSON formatter output I could reference? A TypeScript type would be ideal but given ESLint is in JavaScript I don't expect there is one.

**Comment by Rec0iL99** [comment, marked as answer]
Hi @ataylorme, I think the main type is the [`LintResult`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a1980b0af2c54994fe9f217ace09afa59c2cf72e/types/eslint/index.d.ts#L1398) type.

**Reply by ataylorme** [reply]
Thanks! I didn't think to check that package. For context, I'm hoping it will help me fix some issues like [this one](https://github.com/ataylorme/eslint-annotate-action/issues/72) where I was guessing at the output
